### PR TITLE
Courseleaf

### PR DESCRIPTION
**Pull request by bam95899**

### CODE REVIEW

File: tinker/faculty_bios/__init__.py [hunk 1]
```diff
@@ -85,6 +85,13 @@ def delete_confirm(self):
         return render_template('faculty-bios/delete-confirm.html')
 
     def new(self):
+        if 'Tinker Faculty Bios - Admin' in session['groups'] or 'Administrators' in session['groups'] \
```
Comment (c-schwarze) [inline tinker/faculty_bios/__init__.py]: This should probably be moved to forms.py

File: tinker/faculty_bios/__init__.py [hunk 2]
```diff
@@ -106,6 +113,14 @@ def faculty_bio_in_workflow(self):
         return render_template('faculty-bios/in-workflow.html')
 
     def edit(self, faculty_bio_id):
+        if 'Tinker Faculty Bios - Admin' in session['groups'] or 'Administrators' in session['groups'] \
```
Comment (c-schwarze) [inline tinker/faculty_bios/__init__.py]: This should also be moved to forms.py

File: tinker/faculty_bios/faculty_bio_controller.py [hunk 3]
```diff
@@ -435,6 +436,15 @@ def update_structure(self, faculty_bio_data, sdata, rform, faculty_bio_id=None):
         else:
             add_data['id'] = None
 
+        # Today's Date: (08/09/18)
+        # This hacky chunk of code is used to update the asset to include the courseleaf-user option.
+        # The courseleaf-user option is a new addition to the form and instead of having to edit and submit every single
```
Comment (c-schwarze) [inline tinker/faculty_bios/faculty_bio_controller.py]: Maybe also mention why we would have to edit and submit every faculty bio. And that is submit in Cascade, not in tinker.

Comment (c-schwarze) [reply tinker/faculty_bios/faculty_bio_controller.py]: And maybe mention that its a placeholder - line 448 sets the data

File: tinker/faculty_bios/forms.py [hunk 4]
```diff
@@ -79,6 +79,7 @@ class FacultyBioForm(Form):
         first = StringField('Faculty first name')
         last = StringField('Faculty last name')
         author_faculty = StringField("Faculty member's username", description="Enter your Bethel username.")
+        courseleaf_user = RadioField('', default='Yes', choices=[('Yes', 'Yes'), ('No', 'No')])
```
Comment (c-schwarze) [inline tinker/faculty_bios/forms.py]: You also need to add the hidden field to the forms. (not for the jev username, but for the other one). Also, make sure to test that for both views (admin and normal person)

File: tinker/templates/faculty-bios/form.html [hunk 5]
```diff
@@ -79,6 +79,19 @@ <h3 id="{{ field.id }}">{{ field.label }}</h3>
                             {% elif field.type == "CKEditorTextAreaField" %}
                                 {{ field(class="ckeditor") }}
 
+                            {% elif field.type == 'RadioField' %}
+                                {% if show_special_admin_view %}
```
Comment (c-schwarze) [inline tinker/templates/faculty-bios/form.html]: i mentioned it earlier, but this check should be done in forms.py

Comment (c-schwarze) [reply tinker/templates/faculty-bios/form.html]: The hidden field should actually handle the value nicely - which means it won't even use this if section "field.type == 'RadioField'"

File: tinker/faculty_bios/forms.py (hunk 4 above)
Comment (bam95899) [reply tinker/faculty_bios/forms.py]: Fixed

File: tinker/faculty_bios/__init__.py (hunk 2 above)
Comment (bam95899) [reply tinker/faculty_bios/__init__.py]: Fixed

File: tinker/faculty_bios/__init__.py (hunk 1 above)
Comment (bam95899) [reply tinker/faculty_bios/__init__.py]: Fixed

File: tinker/faculty_bios/faculty_bio_controller.py (hunk 3 above)
Comment (bam95899) [reply tinker/faculty_bios/faculty_bio_controller.py]: fixed

File: tinker/templates/faculty-bios/form.html (hunk 5 above)
Comment (bam95899) [reply tinker/templates/faculty-bios/form.html]: fixed

**Comment by c-schwarze** [comment]
Looks great

**Pull request merged**